pkg/catalogsourceconfig/catalogsourcebuilder.go

```diff
@@ -28,13 +28,29 @@ func (b *CatalogSourceBuilder) WithTypeMeta() *CatalogSourceBuilder {
 // WithMeta sets basic TypeMeta and ObjectMeta.
 func (b *CatalogSourceBuilder) WithMeta(name, namespace string) *CatalogSourceBuilder {
 	b.WithTypeMeta()
-	b.cs.ObjectMeta = metav1.ObjectMeta{
-		Name:      name,
-		Namespace: namespace,
+	objectMeta := b.cs.GetObjectMeta()
+	if objectMeta == nil {
```

Comment:
I am not clear why you need to check for nil though, or may be this be removed once you addressed @kevinrizza's comments?

Reply:
Because if the object meta was already initialized using the `WithMeta()` call, I don't want to overwrite or lose any information. This check will be moved to `WithVisibilityLabel()`.

Reply:
nit: maybe we can make it a bit more concise?
```
if objectMeta == nil {
  b.cs.ObjectMeta = metav1.ObjectMeta{}
}

b.cs.SetName(name)
b.cs.SetNamespace(namespace)
```